portal-impl/src/com/liferay/portal/systemevent/SystemEventAdvice.java

```diff
@@ -225,6 +233,32 @@ protected long getCompanyId(ClassedModel classedModel) {
 		return 0;
 	}
 
+	protected String getExternalReferenceCode(ClassedModel classedModel)
+		throws Exception {
+
+		Class<?> modelClass = classedModel.getClass();
+
+		String className = modelClass.getName();
+
+		Method getUuidMethod = null;
+
+		try {
+			getUuidMethod = modelClass.getMethod(
+				"getExternalReferenceCode", new Class<?>[0]);
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+
+			_noUUIDClassNames.add(className);
```

Comment:
I think you should not use the `_noUUIDClassNames` list, as it is for the missing UUIDs